src/libs/actions/Policy.js

```diff
@@ -256,15 +257,28 @@ function createPolicyExpenseChats(policyID, invitedEmailsToAccountIDs, betas) {
         reportCreationData: {},
     };
 
-    // If the user is not in the beta, we don't want to create any chats
-    if (!Permissions.canUsePolicyExpenseChat(betas)) {
-        return workspaceMembersChats;
-    }
+    const optimisticPersonalDetails = {};
+    const successPersonalDetails = {};
 
     _.each(invitedEmailsToAccountIDs, (accountID, email) => {
         const cleanAccountID = Number(accountID);
         const login = OptionsListUtils.addSMSDomainIfPhoneNumber(email);
 
+        if (_.isEmpty(personalDetails[accountID])) {
+            optimisticPersonalDetails[accountID] = {
+                login,
+                accountID,
+                avatar: UserUtils.getDefaultAvatarURL(accountID),
+                displayName: login,
+            };
+            successPersonalDetails[accountID] = null;
+        }
+
+        // If the user is not in the beta, we don't want to create any chats
+        if (!Permissions.canUsePolicyExpenseChat(betas)) {
+            return;
+        }
```

Comment:
Why did you move this block to inside the loop?

Reply:
@fabioh8010, there is a reason. The app iterates `invitedEmailsToAccountIDs`, on each iteration it should decide now: should create a new local personal details?; should remove this local personal details after?.
I moved this block to inside the loop, because I do not want to iterate `invitedEmailsToAccountIDs` twice. I am going to do it during one loop.

Reply:
So it's just to prevent the next call `const oldChat = ReportUtils.getChatByParticipantsAndPolicy([sessionAccountID, cleanAccountID], policyID);`, but it will push some data to `workspaceMembersChats` anyway, right?

Reply:
Yes, it is to prevent the logic that requires a specific `beta`. The app should always push to `workspaceMembersChats` the next fields:
```javascript
workspaceMembersChats.onyxOptimisticData.push({
     onyxMethod: Onyx.METHOD.MERGE,
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     value: optimisticPersonalDetails,
});
workspaceMembersChats.onyxSuccessData.push({
     onyxMethod: Onyx.METHOD.MERGE,
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     value: successPersonalDetails,
});
```

Reply:
It is a common logic.